src/rdkafka.h

```diff
@@ -6648,8 +6696,139 @@ rd_kafka_DescribeConfigs_result_resources (
         const rd_kafka_DescribeConfigs_result_t *result,
         size_t *cntp);
 
-/**@}*/
 
+/*
+ * DeleteRecords - delete records from partitions
+ *
+ *
+ */
+
+/**
+ * @brief Delete records in topics and partitions as specified by
+ *        the \p offsets
+ *
+ * \p offsets should contain \c topic, \c partition, and \c offset
+ * \c offset is the offset before which the messages will be deleted.
+ * Set \c offset to RD_KAFKA_OFFSET_END (high-watermark) in order to
+ * delete all data in the partition.
+ *
+ * The function will block for at most \p timeout_ms milliseconds while
+ * determining the leader brokers for each partition in \p offsets.
+ *
+ * @param rk Client instance.
+ * @param offsets Offsets to delete up to (exclusive).
+ * @param options Optional admin options, or NULL for defaults.
+ * @param rkqu Queue to emit result on.
+ *
+ * Supported admin options:
+ *  - rd_kafka_AdminOptions_set_operation_timeout() - default 0
+ *  - rd_kafka_AdminOptions_set_request_timeout() - default socket.timeout.ms
+ *
+ * @remark The call may block for up to
```

Comment:
Yes, it will still block for the duration of `rd_kafka_topic_partition_list_query_leaders`. You mentioned earlier that you were going to implement an async version of that function we could call, but that we could merge this PR in the meantime. Is that API available already?

Reply:
Ah, I see, forgot about that. Let me add a PR for that.

Reply:
@edenhill any progress on the async leader_req API?